packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -859,6 +860,34 @@ class FlutterPlugin implements Plugin<Project> {
         return version2
     }
 
+    private void forceNdkDownload(Project gradleProject, String flutterSdkRootPath) {
+        // If the project is already configuring a native build, we don't need to do anything.
+        Boolean forcingNotRequired = gradleProject.android.externalNativeBuild.cmake.path != null
+        if (forcingNotRequired) {
+            return
+        }
+
+        // Otherwise, point to an empty CMakeLists.txt, and ignore associated warnings.
+        gradleProject.android {
+            externalNativeBuild {
+                cmake {
+                    // Respect the existing configuration if it exists - the NDK will already be
+                    // downloaded in this case.
+                    path = flutterSdkRootPath + "/packages/flutter_tools/gradle/src/main/groovy/CMakeLists.txt"
+                    //arguments "-Wno-dev"
+                }
+            }
+
+            defaultConfig {
+                externalNativeBuild {
+                    cmake {
+                        arguments("-Wno-dev", "--no-warn-unused-cli")
```

Comment:
add a comment explaining why these flags and what would cause them to change.

Reply:
Also if there is a way we can share these flags with some other code where we define similar flags lets consider that.